sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/NoSuchItemException.scala

```diff
@@ -29,18 +29,24 @@ import org.apache.spark.sql.types.StructType
  * Thrown by a catalog when an item cannot be found. The analyzer will rethrow the exception
  * as an [[org.apache.spark.sql.AnalysisException]] with the correct position information.
  */
-class NoSuchDatabaseException(
-    val db: String) extends NoSuchNamespaceException(s"Database '$db' not found")
+case class NoSuchDatabaseException(db: String)
+  extends AnalysisException(s"Database '$db' not found")
```

Comment:
Scala disallows a case class to inherit another case class.

Reply:
The change in exception hierarchy is user-visible though. Is this really the only way to fix it vs fixing the issue with copy() directly or something?

Reply:
I can't think of a better way. AFAIK it's an ill pattern to extend a case class in Scala.